en/services/opendroneid.md

```diff
@@ -46,27 +51,53 @@ Basic ID | Basic ID | [OPEN_DRONE_ID_BASIC_ID](../messages/common.md#OPEN_DRONE_
+> **Note** The raw byte layout of the MAVLink messages is not exactly the same as what a RID Bluetooth/Wi-Fi transmitter component will broadcast over the air.
   Slight compression is applied.
   Example code for this conversion can be found in the project: [Open Drone ID Core C Library](https://github.com/opendroneid/opendroneid-core-c).
 
-The [Open Drone ID Core C Library](https://github.com/opendroneid/opendroneid-core-c) contains code for decoding the MAVLink messages and "compressing" the data into data structures for transmission over Bluetooth or Wi-Fi (or vice-versa).
-
-The standards requires that the Location message is broadcast/published at least once per second.
-The rest of the messages must be broadcast/published once per 3 seconds (requirements from local legislation might be different.
-Please see [here](https://github.com/opendroneid/opendroneid-core-c#comparison)).
-Not all message types are mandatory to broadcast.
+The [Open Drone ID Core C Library](https://github.com/opendroneid/opendroneid-core-c) contains code for decoding the MAVLink messages and "compressing" the data into data structures for broadcast over Bluetooth or Wi-Fi (or vice-versa for reception).
 
 The standards do not impose any requirements for a drone to be capable of receiving drone ID messages, nor any requirements for reacting to their content (requirements from local legislation might be different).
 
-An example Android receiver implementation for broadcast drone ID messages is available here: [OpenDroneID Android receiver application](https://github.com/opendroneid/receiver-android).
+An example Android receiver implementation for broadcast drone ID messages is [available here](https://play.google.com/store/apps/details?id=org.opendroneid.android_osm) ([source codes](https://github.com/opendroneid/receiver-android)).
 
 Code related to (Internet) Network Remote ID can be found in the [InterUSS Project](https://github.com/interuss) and https://github.com/uastech/standards (Unofficial reference for UAS-related APIs).
 
+## Message update rates {#update_rates}
+
+The standards require that the Location message is broadcast/published from the RID transmitter component at least once per second.
+The rest of the messages must be broadcast/published once per 3 seconds (the rules in some regions have tightened this requirement to 1 second for also the Basic ID and the System message.
+Please see [here](https://github.com/opendroneid/opendroneid-core-c#comparison)).
+
+Not all message types or fields within a message type are mandatory to broadcast/publish.
+The mandatory message set vary from region to region.
+Please see the [summary](https://github.com/opendroneid/opendroneid-core-c#comparison).
+
+The standards require that the data contained in any broadcasted/published message (on the air/internet) is not older than 1 second (e.g. the location data, timestamps etc.).
```

Comment:
So for messages published every 3 seconds the data must not be more than 1 second old?

Reply:
Sorry, that is actually wrong. Good that you pointed out this. It only applies to the fields in the Location message. I will correct this.

I now realise that the standards are actually missing a requirement for how old the operator location/altitude and the full timestamp in the System message must be. I think we all just implicitly assumed that the 1 second freshness rule applied to fields in all messages.

I had already written the below when I found out the above. I will keep it here just for educational purposes.

Originally, the messages that only require 3 seconds between them, all contained fairly static data. The UA ID, the operator ID etc. never changes. The exception is the System message, which contains the operator location. I assume that it was decided way back (even before I joined the ASTM standard working group) that it is enough to update the operator location every third second.

However, the argument might still go that internally in the UAS, the GCS must send the operator location once per second to the RID transmitter component so that the operator location that is transmitted on the air is never older than one second (even though that only happens every third second).

A very last minute change in the ASTM F3211-22 update was done earlier this year, to meet the FAA requirements. A full timestamp field had to be added somewhere and the only place it would fit was in the System message. However, for the US, the MoC document (ASTM F3586-22) had already overriden the 3 second requirement and mandated 1 second refresh cycles for the Basic ID, the Location and the System message. I.e. for the US, this is quite clear (1 second for everything) and for Europe, the System timestamp is not a mandatory field, so sending System every third second doesn't break any requirement.

Reply:
The MOC seems pretty clear, but I think most explicitly in the testing section, specifically:

> 8.6.2.4 Pass Criteria—For each message type, the time interval between two messages is calculated, and the interval is compared against the maximum as specified in Specification F3411, Subsection 5.4.4.1, and as amended by MOC 5.1.3.8. The interval shall be less than or equal to the allowed time (1 second). In summary, all fields in Table 2 shall be received at least every second.

Reply:
Also 

> 8.9.5.2 Pass Criteria—Each increment shall receive the required messages at least every second for 9 seconds out of 10 seconds and no more than four increments may fail this ratio, and no two failures may be adjacent to one another.

Reply:
The problem, which I only just now discovered, is that you can be compliant with all of the above by just sending the same non-changing data at a high rate. 

Clearly the intention is to get updated operator location and timestamp data but none of the above require the data to be fresh. Only that the transmitter component must ping out whatever (old) data it happens to have once per second.

This is clearly an oversight that no-one discovered during the standardization work.

Reply:
Isn't 6.1.3 pretty clear?

> Additionally, the requirement in MOC 5.1.3.8, following Part 89, effectively mandates that all three required messages are transmitted within 1 second, with this method resulting in **the system timestamp being associated with all required data elements is never older than 1 second.**

Reply:
Those two places might "save" us here but they aren't exactly crystal clear requirements.

@gabrielcox: Do you have any opinions on this? Clearly the intention in the back of the head of all of us was that both GCS location and the system timestamp should be refreshed once per second but we somehow failed to document it clearly. See the discussion above.

Reply:
Gabriel and I had a chat about this and agreed that the the various places that Tim points out above are enough to cover things. The FAA rule is very clear on this topic so there isn't really any incentive for anyone to try to nitpick through the exact wording in the MoC and claim that they can be compliant without having the fields in the System message less than a second old.

[§89.310 h (4)](https://www.regulations.gov/document/FAA-2019-1100-53264): The unmanned aircraft must broadcast the latitude, longitude, and geometric altitude of the unmanned aircraft and its control station no later than 1.0 seconds from the time of measurement to the time of broadcast.

I will update the documentation accordingly and mark this as resolved.